okhttp/src/main/java/okhttp3/internal/platform/android/Android10SocketAdapter.kt

```diff
@@ -54,14 +56,19 @@ class Android10SocketAdapter : SocketAdapter {
     hostname: String?,
     protocols: List<Protocol>
   ) {
-    socketFactory.setUseSessionTickets(sslSocket, true)
+    try {
+      socketFactory.setUseSessionTickets(sslSocket, true)
 
-    val sslParameters = sslSocket.sslParameters
+      val sslParameters = sslSocket.sslParameters
 
-    // Enable ALPN.
-    sslParameters.applicationProtocols = Platform.alpnProtocolNames(protocols).toTypedArray()
+      // Enable ALPN.
+      sslParameters.applicationProtocols = Platform.alpnProtocolNames(protocols).toTypedArray()
 
-    sslSocket.sslParameters = sslParameters
+      sslSocket.sslParameters = sslParameters
+    } catch (iae: IllegalArgumentException) {
+      // probably java.lang.IllegalArgumentException: Invalid input to toASCII from IDN.toASCII
+      throw IOException("Android internal error", iae)
```

Comment:
@swankjesse Thoughts on wording here?

Reply:
Also swallowing here would allow request to proceed, but with HTTP/1.1

Reply:
I like this.